### DISCUSSION TITLE
Don't understand how to configure RPC with the new release ;(

### DISCUSSION THREAD

**Discussion by Guesch** [haugene/docker-transmission-openvpn#2386] [General]
Greetings,

I have no doubt that the issue is between my chair and my screen but unfortunately I can't seem to find the solution on my own :(

RPC was working prior to the update but now it doesn't recognize my login/password anymore.
In the setting.json "rpc-authentication-required" is set to true, rpc-username & rpc-password to whatever I want. But when I try to connect it says that the same login & password are incorrect. 
When I relaunch the docker, the RPC settings are reset.. 

I can see that a new fancy "transmission-credentials.txt" file appeared, so I tried to add the login in the first line and the password in the 2nd line; but same happens: Transmission WebUI doesn't recognize them and when I restart the docker the text file is empty again.

Please note that I didn't change anything in the docker compose, except to add /config path (Everything I'm describing above is happening in the new /config path.. I deleted the old transmission-home from the data folder to start clean)

Any help will be super appreciated!

Thank you!

**Comment by GDXbsv** [comment]
It is the same for me

Reason is:  the container started to override all the time, even i did not set them in my compose file:
 ```
 transmission-vpn  | Overriding rpc-password because TRANSMISSION_RPC_PASSWORD is set to [REDACTED]
transmission-vpn  | Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
transmission-vpn  | Overriding rpc-username because TRANSMISSION_RPC_USERNAME is set to 
transmission-vpn  | sed'ing True to true
transmission-vpn  | STARTING TRANSMISSION
 ```
 
 I found only one solution. I did fix them in compose file:
 ```
     environment:
      - TRANSMISSION_RPC_USERNAME=transmission
      - TRANSMISSION_RPC_PASSWORD="YOUR PASSWORD"
      - TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=true

 ```

**Comment by toddrob99** [comment]
I am submitting an issue for this. It appears #2321 broke our use case where the credentials are stored in transmission-credentials.txt but not in environment variables (I removed from environment variables because the credentials were stored in the txt file). Now the logic does not persist pre-existing values in the txt file.

**Reply by ColinHebert** [reply]
Not quite, as you can see in #2321, the code was already doing this
```
echo "${TRANSMISSION_RPC_USERNAME}" > /config/transmission-credentials.txt
echo "${TRANSMISSION_RPC_PASSWORD}" >> /config/transmission-credentials.txt
```

Which effectively erases the content of `transmission-credentials.txt` if the creds aren't set as environment variables.
The PR you pointed out has its own issues, but isn't the cause of the problem you are mentioning.

**Comment by toddrob99** [comment]
https://github.com/haugene/docker-transmission-openvpn/issues/2387
